tests/Microsoft.Experimental.Collections.Tests/Microsoft/Collections/Extensions/OrderedDictionaryTests.cs

```diff
@@ -0,0 +1,276 @@
+            for (var i = 0; i < 5; ++i)
+            {
+                test.Add(i.ToString(), i);
+            }
+
+            // Tests Count
+            Assert.Equal(5, test.Count);
+            // Tests Keys.Count
+            Assert.Equal(5, test.Keys.Count);
+            // Tests Values.Count
+            Assert.Equal(5, test.Values.Count);
+            for (var i = 0; i < 5; ++i)
+            {
+                // Tests ContainsKey
+                Assert.True(test.ContainsKey(i.ToString()));
+                // Tests Key Indexer
+                Assert.Equal(i, test[i.ToString()]);
+                // Tests Array Indexer
+                Assert.Equal(i, test[i]);
+                // Tests IndexOf
+                Assert.Equal(i, test.IndexOf(i.ToString()));
+            }
+
+            // Tests removing
+            Assert.True(test.Remove("1"));
+            Assert.Equal(4, test.Count);
+            Assert.Equal(4, test.Keys.Count);
+            Assert.Equal(4, test.Values.Count);
+            Assert.False(test.ContainsKey("1"));
+            Assert.True(test.ContainsKey("0"));
+            Assert.Equal(0, test["0"]);
+            Assert.Equal(0, test.IndexOf("0"));
+            Assert.Equal(0, test[0]);
+            Assert.True(test.ContainsKey("2"));
+            Assert.Equal(2, test["2"]);
+            Assert.Equal(1, test.IndexOf("2"));
+            Assert.Equal(2, test[1]);
+            Assert.True(test.ContainsKey("3"));
+            Assert.Equal(3, test["3"]);
+            Assert.Equal(2, test.IndexOf("3"));
+            Assert.Equal(3, test[2]);
+            Assert.True(test.ContainsKey("4"));
+            Assert.Equal(4, test["4"]);
+            Assert.Equal(3, test.IndexOf("4"));
+            Assert.Equal(4, test[3]);
+
+            // Tests inserting
+            test.Insert(1, "1", 1);
+            Assert.Equal(5, test.Count);
+            Assert.Equal(5, test.Keys.Count);
+            Assert.Equal(5, test.Values.Count);
+            Assert.True(test.ContainsKey("0"));
+            Assert.Equal(0, test["0"]);
+            Assert.Equal(0, test.IndexOf("0"));
+            Assert.Equal(0, test[0]);
+            Assert.True(test.ContainsKey("1"));
+            Assert.Equal(1, test["1"]);
+            Assert.Equal(1, test.IndexOf("1"));
+            Assert.Equal(1, test[1]);
+            Assert.True(test.ContainsKey("2"));
+            Assert.Equal(2, test["2"]);
+            Assert.Equal(2, test.IndexOf("2"));
+            Assert.Equal(2, test[2]);
+            Assert.True(test.ContainsKey("3"));
+            Assert.Equal(3, test["3"]);
+            Assert.Equal(3, test.IndexOf("3"));
+            Assert.Equal(3, test[3]);
+            Assert.True(test.ContainsKey("4"));
+            Assert.Equal(4, test["4"]);
+            Assert.Equal(4, test.IndexOf("4"));
+            Assert.Equal(4, test[4]);
+
+            // Tests OrderedDictionary Enumeration
+            var temp = 0;
+            foreach (var pair in test)
```

Comment:
nit: avoid using var here (and for `var temp`) since it isn't clear from reading the right hand side what the type is.

Reply:
:) It's hard to break old habits.